features/static_primary.feature

```diff
@@ -0,0 +1,20 @@
+Feature: static primary
+  We should check that static primary behavior is safe
+
+  Scenario: check static primary config in dcs blocks replica from starting
+    Given I start postgres0 as static primary
+    Then postgres0 is a leader after 10 seconds
+    And there is a non empty initialize key in DCS after 15 seconds
+    When I issue a PATCH request to http://127.0.0.1:8008/config with {"ttl": 20, "loop_wait": 2, "synchronous_mode": true}
+    Then I receive a response code 200
+    When I start postgres1 with a configured static primary will not boot after 20 seconds
+    And I start postgres2 with a configured static primary will not boot after 20 seconds
+    And "sync" key not in DCS after waiting 20 seconds
+    And "members/postgres1" key not in DCS after waiting 10 seconds
+    And "members/postgres2" key not in DCS after waiting 10 seconds
+
+  Scenario: check removing static primary config from dcs allows replica startup
+    Given I issue a PATCH request to http://127.0.0.1:8008/config with {"static_primary": null}
+    Then "sync" key in DCS has leader=postgres0 after 20 seconds
+    And "members/postgres1" key in DCS has state=running after 10 seconds
+    And "members/postgres2" key in DCS has state=running after 10 seconds
```

Comment:
Since postgres1 and postgres2 are not supposed to be running this scenario will fail and should be removed.

Reply:
This scenarios does indeed succeed, because the Patroni instances are still running. It is only the Postgres processes which are not starting due to the static_primary config (which is intended behavior).

Once that static_primary config is set to null in the DCS, then the Postgres replicas are able to start.

I think this is a point of confusion based on one of your recent comments in ha.py. The replica Patroni instances are not intended to sys.exit. The idea is to just keep Postgres from starting, and the Patroni process stays alive and will continue to check the DCS values periodically based on its control loop.

This will allow for a more flexible (and still safe) operator experience. Basically, cluster operators do not need to synchronize the steps of removing static_primary config and adding of new replicas. The operations are safe to overlap (fewer footguns).